pkg/token/cache.go

```diff
@@ -55,12 +57,16 @@ type LockedTokenCache struct {
 
 	// memoryUsage is the estimated number of bytes used by the cache.
 	memoryUsage int64
+
+	// tokenType is used as a constant label for prometheus metrics
+	tokenType string
 }
 
-func NewLockedTokenCache() *LockedTokenCache {
+func NewLockedTokenCache(tokenType string) *LockedTokenCache {
 	return &LockedTokenCache{
 		cache:       make(map[CacheKey]Token),
 		memoryUsage: 0,
+		tokenType:   tokenType,
 	}
 }
```

Comment:
Possibly better if you have two different functions instead
```suggestion
func NewLockedRoleTokenCache() *LockedTokenCache {
	return &LockedTokenCache{
		cache:       make(map[CacheKey]Token),
		memoryUsage: 0,
		tokenType:   "roletoken",
	}
}
func NewLockedAccessTokenCache() *LockedTokenCache {
	return &LockedTokenCache{
		cache:       make(map[CacheKey]Token),
		memoryUsage: 0,
		tokenType:   "accesstoken",
	}
}
```

so that you can do
```go
	accessTokenCache := NewLockedAccessTokenCache()
	roleTokenCache := NewLockedRoleTokenCache()
```